tests/integration/release/utilities.py

```diff
@@ -0,0 +1,222 @@
+from __future__ import annotations
+
+import dask.dataframe as dd
+import numpy as np
+import pandas as pd
+from pytest_check import check
+from vivarium_testing_utils import FuzzyChecker
+
+from pseudopeople.configuration import Keys
+from pseudopeople.configuration.noise_configuration import NoiseConfiguration
+from pseudopeople.constants.noise_type_metadata import (
+    GUARDIAN_DUPLICATION_ADDRESS_COLUMNS,
+)
+from pseudopeople.dataset import Dataset
+from pseudopeople.noise_entities import NOISE_TYPES
+from pseudopeople.noise_functions import merge_dependents_and_guardians
+from pseudopeople.schema_entities import DATASET_SCHEMAS
+
+
+def run_do_not_respond_tests(
+    prenoised_dataframes: list[pd.DataFrame],
+    noised_datasets: list[Dataset],
+    config: NoiseConfiguration,
+    dataset_name: str,
+    fuzzy_checker: FuzzyChecker,
+) -> None:
+    """Takes a list of prenoised DataFrames and matching noised Datasets and checks that
+    after applying do_not_respond, 1) columns and dtypes were preserved properly and
+    2) the expected number of rows were removed."""
+    expected_noise = config.get_row_probability(dataset_name, NOISE_TYPES.do_not_respond.name)
+
+    # ACS and CPS data are oversampled by a factor of 2 so we apply a baseline probability of
+    # 0.5 for do_not_respond so the data returned is of the expected size
+    # # TODO: [MIC-6002] move this processing step out of noising
+    if dataset_name is not DATASET_SCHEMAS.census.name:  # is acs or cps
+        expected_noise = 0.5 + expected_noise / 2
+
+    numerator = 0
+    denominator = 0
+    for prenoised_dataframe, noised_dataset in zip(prenoised_dataframes, noised_datasets):
+        # Do all the calculations and non-fuzzy checks
+        with check:
+            assert set(noised_dataset.data.columns) == set(prenoised_dataframe.columns)
+        with check:
+            assert (noised_dataset.data.dtypes == prenoised_dataframe.dtypes).all()
+
+        shard_numerator = len(prenoised_dataframe) - len(noised_dataset.data)
+        shard_denominator = len(prenoised_dataframe)
+
+        numerator += shard_numerator
+        denominator += shard_denominator
+
+    # Test that noising affects expected proportion with expected types
+    with check:
+        fuzzy_checker.fuzzy_assert_proportion(
+            name="test_do_not_respond",
+            observed_numerator=numerator,
+            observed_denominator=denominator,
+            # 3% uncertainty on either side
```

Comment:
I can't find where this was written right now but I have a hunch this was meant to be percentage points, i.e. `expected_noise - 0.03, expected_noise + 0.03`.